_includes/selector.html

```diff
@@ -481,37 +490,73 @@
             },
             getpipCmdHtml() {
                 var pip_install = `${this.highlightCmd("pip")} install`;
-                var index_url = `--${this.highlightFlag("extra-index-url")}=https://pypi.nvidia.com`;
                 var cuda_suffix = "-cu12";
                 var indentation = "    ";
 
                 if (this.active_pip_cuda_ver === "11.2 - 11.8") {
                     cuda_suffix = "-cu11";
                 }
 
-                var libraryToPkg = (pkg) => {
-                    pkg = pkg.toLowerCase();
-                    if (pkg === "cucim") return pkg;
-                    if (pkg == "cuspatial/cuproj") return "cuspatial" + cuda_suffix + " cuproj" + cuda_suffix;
-                    return pkg + cuda_suffix;
+                // Change index depending on stable vs nightly for pip
+                // Also add versioning commands for nightly installs so that --pre is unnecessary
+                // This has duplicate code, but makes for easier edits in the future
+                if (this.active_release === "Stable") {
+                    index_url = `--${this.highlightFlag("extra-index-url")}=https://pypi.nvidia.com`;
+                    var libraryToPkg = (pkg) => {
+                        pkg = pkg.toLowerCase();
+                        if (pkg === "cucim") return pkg; // Remove this line in 23.12 when stable cucim is out on our index
+                        if (pkg === "cuspatial/cuproj") return ["cuspatial" + cuda_suffix, "cuproj" + cuda_suffix];
+                        if (pkg === "raft") return ["pylibraft" + cuda_suffix, "raft-dask" + cuda_suffix];
+                        return [pkg + cuda_suffix];
+                    }
+                }
+                else {
+                    index_url = `--${this.highlightFlag("extra-index-url")}=https://pypi.anaconda.org/rapidsai-wheels-nightly/simple`;
+                    cuda_suffix = cuda_suffix + ">={{ site.data.releases.nightly.version }}.0a0,<={{ site.data.releases.nightly.version }}\"";
```

Comment:
Should `<=` be `<`, to prevent stable package installation?
```suggestion
                    cuda_suffix = cuda_suffix + ">={{ site.data.releases.nightly.version }}.0a0,<{{ site.data.releases.nightly.version }}\"";
```

Reply:
I thought so too, but it actually won't resolve at all if you do that (I just double checked to make sure). 

We could do
`...,<23.12.0a$sufficiently-large-number`

Not sure what's less ideal, the $sufficiently-large-number, or having users get a stable 23.12 between 24.02's release and the site update (I think that's the only time it could happen.) It's already a long pip install command so not much lost in changing it...

Reply:
Oof. I'd prefer `<23.12.0rc0` over `<23.12.0a9999`. But it's probably fine to permit stable builds. Can you revert 68a5cc2? Let's just roll with the previous `<=`.

Reply:
cc: @vyasr if you have last minute thoughts here.

Reply:
> having users get a stable 23.12 between 24.02's release and the site update

The user would also have to mix between the stable and nightly index -- I'm not worried about this happening in practice.

Reply:
Maybe we can revert 68a5cc2, merge this, and then revisit it in a new PR for any later improvements. I'd like to make some changes to this page but don't want to introduce conflicts with this PR.

Reply:
Done. Happy to merge so you can get changes in. Will do so in a couple hours just in case there are last minute comments from CCed people/you. Or if you want to get working on other changes asap feel free to merge when you're ready to work.

Reply:
Just seeing this post-merge, but FWIW I'm also not overly concerned about mixing nightlies and stable indexes, so I don't think we need to overengineer the command specifically to avoid the stable release. Also the whole point of the versioning changes was that nightlies are strictly ordered less than stable releases, so if the wrong package gets installed users should be able to use normal pip commands to upgrade without needing to go through manual uninstall/install cycles.